accepted/future-releases/records/records-feature-specification.md

```diff
@@ -122,6 +122,8 @@ not captured by the grammar. It is a compile-time error if a record has any of:
 
 *   Only one positional field and no trailing comma.
 
+*   No fields and a trailing comma. *The expression `(,)` isn't allowed.*
```

Comment:
My guess is that this is (or should be) captured by the grammar. 
In the rest of our grammar, an optional trailing colon is usually handled as:
```dart
collection ::= startToken elements? endToken
elements ::= element (',' element)* ','?
```

The grammar above doesn't actually allow `()` as a record. And it allows `(e)`, which is ambiguous.
Having a prose section saying that it isn't allowed is technically enough, but not how we usually do it.

How about:
```abnf
record ::= 'const'? '(' recordFields? ')'

recordFields ::= expression (',' | (',' recordField)+ ','?)
               | namedRecordField ( ',' recordField )* ','?

recordField ::= expression | namedRecordField
namedRecordField ::= identifier ':' expression
```

An alternative phrasing would be:
```dart
recordFields ::= expression ','
               | namedRecordField ','?,
               | recordField (',' recordField)+ ','?
```

Reply:
> My guess is that this is (or should be) captured by the grammar.

I thought about that.

> Having a prose section saying that it isn't allowed is technically enough, but not how we usually do it.

Yes, and the implementers tell me often that they don't like the way we currently do it. :)

Every time I try to sort out `type`, `typeNotVoid`, `typeNotFunction`, and `typeNotVoidNotFunction`, I get a little sadder.